sympy/integrals/laplace.py

```diff
@@ -997,6 +1000,45 @@ def _laplace_apply_simple_rules(f, t, s):
     return None
 
 
+def _piecewise_to_heaviside(f, t):
+    """
+    This function converts a Piecewise expression to an expression written
+    with Heaviside. It is not exact, but valid in the context of the Laplace
+    transform.
+    """
+    tr = Dummy('tr', real=True)
+    x = piecewise_exclusive(f.subs(t, tr))
+    r = S.Zero
+    for fn, cond in x.args:
+        # Here we do not need to do many checks because piecewise_exclusive
+        # has a clearly predictable output. However, if any of the conditions
+        # is not relative to t, this function just returns the input argument.
+        if isinstance(cond, Relational) and tr in cond.args:
+            r = r + Heaviside(cond.gts - cond.lts)*fn
+        elif (
+              isinstance(cond, Or)):
+            for c2 in cond.args:
+                if tr in c2.args:
+                    r = r + Heaviside(c2.gts - c2.lts)*fn
+                else:
+                    return f
+        elif (    # And(t <= a, t > b)
+                isinstance(cond, And)
+                and len(cond.args) == 2):
+            c0, c1 = cond.args
+            if tr in c0.args and tr in c1.args:
+                if '>' in c0.rel_op:
+                    c0, c1 = c1, c0
+                r = (
+                    r + (Heaviside(c1.gts-c1.lts) -
+                         Heaviside(c0.lts-c0.gts))*fn)
+            else:
+                return f
+        else:
+            return f
+    return r.subs(tr, t)
```

Comment:
```suggestion
    if not t.is_real:
        r= Dummy('r', real=True)
        return _piecewise_to_heaviside(f.xreplace({t: r}), r).xreplace({r: t}):
    r = []
    for fn, cond in x.args:
        # Here we do not need to do many checks because piecewise_exclusive
        # has a clearly predictable output. However, if any of the conditions
        # is not relative to t, this function just returns the input argument.
        if isinstance(cond, Relational) and t in cond.args:
            r.append(Heaviside(cond.gts - cond.lts)*fn)
        elif isinstance(cond, Or):
            for c in cond.args:
                if t in c.args:
                    r.append(Heaviside(c.gts - c.lts)*fn)
                else:
                    return f
        elif (  # And(t <= a, t > b),And(t <= a, t >= b),And(t < a, t >= b), or And(t < a, t > b)
                isinstance(cond, And)
                and len(cond.args) == 2):
            c0, c1 = cond.args
            if t in c0.args and t in c1.args:
                # put t on left
                if c0.lhs != t:
                    c0 = c0.reversed
                if c1.lhs != t:
                    c1 = c1.reversed
                # put t < a or t <= a as c0
                if '>' in c0.rel_op:
                    c0, c1 = c1, c0
                # check that they are as expected
                if '<' in c0.rel_op and '>' in c1.rel_op:
                    r.append((Heaviside(c1.gts - c1.lts) - Heaviside(c0.lts - c0.gts))*fn)
                else:
                    return f
            else:
                return f
        else:
            return f
    return Add(*r)
```

Reply:
Thanks! I have done it a tiny bit differently, see next push.